lib/Tools/circt-bmc/ExternalizeRegisters.cpp

```diff
@@ -96,22 +96,16 @@ void ExternalizeRegistersPass::runOnOperation() {
           }
           mlir::Attribute initState;
           if (auto initVal = regOp.getInitialValue()) {
-            // Find the seq.initial op where the initial value is defined and
-            // fetch the operation inside that defines the value
-            auto initialOp =
-                regOp.getInitialValue().getDefiningOp<seq::InitialOp>();
-            if (!initialOp) {
+            // Find the constant op that defines the reset value in an initial
+            // block (if it exists)
+            if (isa<BlockArgument>(initVal) ||
+                !isa<seq::InitialOp>(initVal.getDefiningOp())) {
```

Comment:
Bit unfortunate that `unwrapImmutableValue` asserts that the passed value comes from an initial op instead of just returning a null value.

Nit: this seems a bit more readable to me:
```suggestion
            if (!initVal.getDefiningOp<seq::InitialOp>()) {
```

Reply:
Agreed, it would be nice to move this inside the function at some point, that feels more in line with what we see elsewhere.

Agreed that that's nicer, thanks!